diff --git a/components/esp_common/include/esp_err.h b/components/esp_common/include/esp_err.h
index 105723976..3eac9b6d9 100644
--- a/components/esp_common/include/esp_err.h
+++ b/components/esp_common/include/esp_err.h
@@ -106,7 +106,7 @@ void _esp_error_check_failed_without_abort(esp_err_t rc, const char *file, int l
         esp_err_t __err_rc = (x);                                       \
         (void) sizeof(__err_rc);                                        \
     } while(0)
-#elif defined(CONFIG_COMPILER_OPTIMIZATION_ASSERTIONS_SILENT)
+#elif 0 // defined(CONFIG_COMPILER_OPTIMIZATION_ASSERTIONS_SILENT)
 #define ESP_ERROR_CHECK(x) do {                                         \
         esp_err_t __err_rc = (x);                                       \
         if (__err_rc != ESP_OK) {                                       \
diff --git a/components/freertos/include/freertos/FreeRTOSConfig.h b/components/freertos/include/freertos/FreeRTOSConfig.h index bd3c72db5..843eb8d08 100644
--- a/components/freertos/include/freertos/FreeRTOSConfig.h
+++ b/components/freertos/include/freertos/FreeRTOSConfig.h
@@ -123,13 +123,11 @@ int xt_clock_freq(void) __attribute__((deprecated));
 #define configASSERT(a) /* assertions disabled */
 #elif defined(CONFIG_FREERTOS_ASSERT_FAIL_PRINT_CONTINUE)
 #define configASSERT(a) if (!(a)) {                                     \
-        ets_printf("%s:%d (%s)- assert failed!\n", __FILE__, __LINE__,  \
-                   __FUNCTION__);                                       \
+        ets_printf("(%s)- assert failed!\n", __FUNCTION__);             \
     }
 #else /* CONFIG_FREERTOS_ASSERT_FAIL_ABORT */
 #define configASSERT(a) if (!(a)) {                                     \
-        ets_printf("%s:%d (%s)- assert failed!\n", __FILE__, __LINE__,  \
-                   __FUNCTION__);                                       \
+        ets_printf("(%s)- assert failed!\n", __FUNCTION__);             \
         abort();                                                        \
         }
 #endif
