diff --git a/components/driver/sdspi_host.c b/components/driver/sdspi_host.c
index 6f044d143..8adbb52db 100644
--- a/components/driver/sdspi_host.c
+++ b/components/driver/sdspi_host.c
@@ -462,6 +463,7 @@ esp_err_t sdspi_host_start_command(int slot, sdspi_hw_cmd_t *cmd, void *data,
         if (flags & SDSPI_CMD_FLAG_WRITE) {
             ret = start_command_write_blocks(slot, cmd, data, data_size, multi_block, stop_transmission);
         } else {
+            go_idle_clockout(slot);
             ret = start_command_read_blocks(slot, cmd, data, data_size, stop_transmission);
         }
     } else {