diff --git a/components/driver/sdspi_host.c b/components/driver/sdspi_host.c
index 4975b857de..c31286047b 100644
--- a/components/driver/sdspi_host.c
+++ b/components/driver/sdspi_host.c
@@ -622,7 +622,9 @@ static esp_err_t poll_cmd_response(int slot, sdspi_hw_cmd_t *cmd)
  */
 static esp_err_t start_command_read_blocks(int slot, sdspi_hw_cmd_t *cmd,
         uint8_t *data, uint32_t rx_length)
-{
+{
+    go_idle_clockout(slot);
+
     bool need_stop_command = rx_length > SDSPI_MAX_DATA_LEN;
     spi_transaction_t* t_command = get_transaction(slot);
     *t_command = (spi_transaction_t) {
