diff --git a/components/driver/sdspi_host.c b/components/driver/sdspi_host.c
index a42c23393..6519e66aa 100644
--- a/components/driver/sdspi_host.c
+++ b/components/driver/sdspi_host.c
@@ -652,6 +652,7 @@ static esp_err_t shift_cmd_response(sdspi_hw_cmd_t* cmd, int sent_bytes)
 static esp_err_t start_command_read_blocks(int slot, sdspi_hw_cmd_t *cmd,
         uint8_t *data, uint32_t rx_length, bool need_stop_command)
 {
+    go_idle_clockout(slot);
     spi_transaction_t* t_command = get_transaction(slot);
     *t_command = (spi_transaction_t) {
         .length = (SDSPI_CMD_R1_SIZE + SDSPI_RESPONSE_MAX_DELAY) * 8,